http_post.go

```diff
@@ -108,30 +134,54 @@ func (g *httpPostGroup) DeleteGroup(d edge.DeleteGroupMessage) (edge.Message, er
 	return d, nil
 }
 
-func (n *HTTPPostNode) postRow(row *models.Row) {
+func (n *HTTPPostNode) doPost(row *models.Row) int {
+	resp, err := n.postRow(row)
+	if err != nil {
+		n.diag.Error("failed to POST data", err)
+		return 0
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode/100 != 2 {
+		var err error
+		if n.c.CaptureResponseFlag {
+			var body []byte
+			body, err = ioutil.ReadAll(resp.Body)
+			if err == nil {
+				// Use the body content as the error
+				err = errors.New(string(body))
+			}
+		} else {
+			err = errors.New("unknown error, use .captureResponse() to capture the HTTP response")
+		}
+		n.diag.Error("POST returned non 2xx status code", err, keyvalue.KV("code", strconv.Itoa(resp.StatusCode)))
```

Comment:
I just now understood what you were saying here https://github.com/influxdata/kapacitor/pull/1535#discussion_r139201996 about making everything strings.